contracts/src/v0.8/PermissionedForwardProxy.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "./ConfirmedOwner.sol";
+
+/**
+ * @title PermissionedForwardProxy
+ * @notice This proxy is used to forward calls from sender to target. It maintains
+ * a permission list to check which sender is allowed to call which target
+ */
+contract PermissionedForwardProxy is ConfirmedOwner {
+  using Address for address;
+
+  error PermissionNotSet();
+
+  event PermissionSet(address indexed sender, address target);
+  event PermissionRemoved(address indexed sender);
+
+  mapping(address => address) private s_forwardPermissionList;
```

Comment:
I'm curious how useful the `target` really is in this contract? Could we not just manage a permissioned sender list? Then the permissioned sender could use the forwarder contract to communicate with anyone. ex... 
```solidity
mapping(address => bool) private s_forwardPermissionList;
```

Reply:
Discussed offline, this helps segregate the permission on the callee side (feeds)